TheMovieDB/MoviesViewController.swift

```diff
@@ -0,0 +1,193 @@
+        searchBarMovie.showsCancelButton = true
+        searchingNow = true
+        
+        tableMovies.reloadData()
+    }
+    
+    func searchBarCancelButtonClicked(_ searchBar: UISearchBar) {
+        
+        searchBarMovie.showsCancelButton = false
+        searchingNow = false
+        
+        searchBarMovie.endEditing(true)
+        searchBarMovie.text = ""
+        
+        arrMoviesSearch = []
+        currentPageSearch = 0
+        
+        tableMovies.scrollIndicatorInsets = UIEdgeInsets.zero
+        tableMovies.reloadData()
+    }
+    
+    func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        
+        querySearch = searchBar.text!
+        searchMovie(query: querySearch, page: 1)
+        searchBarMovie.endEditing(true)
+    }
+    
+    func scrollViewWillBeginDragging(_ scrollView: UIScrollView) {
+        searchBarMovie.endEditing(true)
+    }
+    
+    //MARK: UITableViewDataSource
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if searchingNow {
+            return arrMoviesSearch.count
+        } else {
+            return arrMovies.count
+        }
+        
+    }
+    
+    func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        let lastElement = arrMovies.count - 1
+        if indexPath.row == lastElement {
+            if searchingNow {
+                if currentPageSearch < totalPagesSearch {
+                    currentPageSearch += 1
+                    searchMovie(query: querySearch, page: currentPageSearch)
+                }
+            } else {
+                if currentPage < totalPages {
+                    currentPage += 1
+                    loadMovies(page: currentPage)
+                }
+            }
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        
+        let cell = tableView.dequeueReusableCell(withIdentifier: "CellMovie", for: indexPath as IndexPath) as! MovieTableViewCell
+        
+        let movie:Movie = getMovie(row: indexPath.row)
+        cell.setData(movie: movie)
+        
+        cell.backgroundColor = UIColor.clear
+        cell.selectionStyle = UITableViewCellSelectionStyle.none
+        
+        return cell
+    }
+    
+    //MARK: UITableViewDelegate
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        
+        let movie:Movie = getMovie(row: indexPath.row)
+        tableView.deselectRow(at: indexPath as IndexPath, animated: true)
+        
+        let viewDetail:MovieDetailViewController = self.storyboard!.instantiateViewController(withIdentifier: "movieDetail") as! MovieDetailViewController;
+        viewDetail.movieData = movie
+        
+        self.navigationController?.pushViewController(viewDetail, animated: true)
+    }
+    
+    //MARK: Get movie indexPath.row
+    func getMovie(row:Int) -> Movie {
```

Comment:
It is not usual to use the word `get` in objective C or Swift method names. This looks Javaish